website_seo_redirection/models/website_seo_redirection.py

```diff
@@ -42,6 +54,21 @@ def _check_destination(self):
         self._url_format_check("destination")
 
     @api.multi
+    @api.constrains("origin", "destination")
+    def _check_not_recursive(self):
+        """Avoid infinite loops."""
+        for redirection in self:
+            origins = {redirection.origin}
+            while redirection:
+                redirection = self.search([
+                    ("origin", "=", redirection.destination),
+                ])
+                if redirection.origin in origins:
+                    raise ValidationError(
+                        _("Recursive redirection is forbidden."))
+                origins.add(redirection.origin)
```

Comment:
But the loop won't terminate until after `False.origin`, which would be yielding an `AttributeError` no?